pkg/controllers/readiness/wellknown_ready_controller.go

```diff
@@ -0,0 +1,352 @@
+package readiness
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	configv1lister "github.com/openshift/client-go/config/listers/config/v1"
+	routeinformer "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+	routev1lister "github.com/openshift/client-go/route/listers/route/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/transport"
+)
+
+var kasServicePort int
+
+func init() {
+	(&sync.Once{}).Do(func() {
+		var err error
+		kasServicePort, err = strconv.Atoi(os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS"))
+		if err != nil {
+			klog.Warningf("Defaulting KUBERNETES_SERVICE_PORT_HTTPS to 443 due to parsing error: %v", err)
+			kasServicePort = 443
+		}
+	})
+}
+
+type wellKnownReadyController struct {
+	serviceLister  corev1lister.ServiceLister
+	endpointLister corev1lister.EndpointsLister
+	operatorClient v1helpers.OperatorClient
+	authLister     configv1lister.AuthenticationLister
+	routeLister    routev1lister.RouteLister
+}
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"WellKnownEndpointDegraded",
```

Comment:
You're not actually setting this value anywhere

Reply:
Since you don't set it to true, move it to Stale conditions

Reply:
i just forgot to remove it, thanks

Reply:
You still need to move `WellKnownEndpointDegraded` among the other stale conditions in the stale condition controller, this one is the only pre-existing type